litr/src/main/java/com/linkedin/android/litr/utils/CodecUtils.java

```diff
@@ -172,99 +175,129 @@ public static int getHighestSupportedProfile(@NonNull String mimeType, boolean i
      * @param codecFormatNotFoundError message to provide in {@link TrackTranscoderException} if codec could not be found by format
      * @param codecConfigurationError message to provide in {@link TrackTranscoderException} if codec could not configured
      * @param fallbackToGetCodecByType flag indicating if older getCodecByType API should be used if getting by format fails
-     * @return configured instance of {@link MediaCodec}, null if codec could not be found or configured
+     * @return configured instance of {@link MediaCodec}, or a {@link TrackTranscoderException} will be thrown
      */
-    @Nullable
+    @NonNull
     public static MediaCodec getAndConfigureCodec(@NonNull MediaFormat mediaFormat,
                                                   @Nullable Surface surface,
                                                   boolean isEncoder,
                                                   @NonNull TrackTranscoderException.Error codecNotFoundError,
                                                   @NonNull TrackTranscoderException.Error codecFormatNotFoundError,
                                                   @NonNull TrackTranscoderException.Error codecConfigurationError,
                                                   boolean fallbackToGetCodecByType) throws TrackTranscoderException {
-        MediaCodec mediaCodec = null;
         try {
-            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {
-                mediaCodec = getAndConfigureCodecByConfig(mediaFormat, surface, isEncoder);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
From the functionality perspective this is a pretty big and important change. I usually do these in a way that allows A/B testing. For example, here I would recommend making your entire new logic a new set of methods and calling the new equivalent of this method from `MediaCodecEncoder` and `MediaCodecDecoder` depending on a flag we pass in in the constructor, similar to what we do with `fallbackToGetCodecByType`

Reply:
@izzytwosheds could you please suggest a name for this flag?

Reply:
filterByTypeAndFormat? lookupByTypeAndFormat?

Reply:
Done